packages/tx/src/legacyTransaction.ts

```diff
@@ -111,7 +111,8 @@ export default class Transaction extends BaseTransaction<Transaction> {
     this.gasPrice = new BN(toBuffer(txData.gasPrice === '' ? '0x' : txData.gasPrice))
 
     if (this.gasPrice.mul(this.gasLimit).gt(MAX_INTEGER)) {
-      throw new Error('gas limit * price overflow')
+      const msg = this._errorMsg('gas limit * gasPrice cannot exceed MAX_INTEGER(2^256-1)')
```

Comment:
```suggestion
      const msg = this._errorMsg('gas limit * gasPrice cannot exceed MAX_INTEGER (2^256-1)')
```

the eip1559Transaction error said `2^256` as opposed `2^256-1` so probably want to align them (also the space after MAX_INTEGER is more understandable imo)